File: ALADDIN_NEW/mobile_apps/ALADDIN_iOS/Shared/Components/Cards/FamilyMemberCard.swift
import SwiftUI

/// 👤 Family Member Card
/// Карточка члена семьи для экрана Family
/// Источник дизайна: HTML .member-card на 02_family_screen.html
struct FamilyMemberCard: View {
    
    // MARK: - Properties
    
    let name: String
    let role: FamilyRole
    let avatar: String
    let status: ProtectionStatus
    let threatsBlocked: Int
    let lastActive: String
    let action: () -> Void
    
    // MARK: - Family Role
    
    enum FamilyRole {
        case parent      // Родитель
        case child       // Ребёнок
        case teenager    // Подросток
        case elderly     // Пожилой
        
        var label: String {
            switch self {
            case .parent: return "Родитель"
            case .child: return "Ребёнок"
            case .teenager: return "Подросток"
            case .elderly: return "Пожилой"
            }
        }
        
        var icon: String {
            switch self {
            case .parent: return "👨‍💼"
            case .child: return "👶"
            case .teenager: return "🧒"
            case .elderly: return "👴"
            }
        }
    }
    
    // MARK: - Protection Status
    
    enum ProtectionStatus {
        case protected   // 🟢 Защищён
        case warning     // ⚠️ Предупреждение
        case danger      // 🔴 Опасность
        case offline     // ⚫ Оффлайн
        
        var label: String {
            switch self {
            case .protected: return "Защищён"
            case .warning: return "Внимание"
            case .danger: return "Угроза"
            case .offline: return "Оффлайн"
            }
        }
        
        var color: Color {
            switch self {
            case .protected: return .successGreen
            case .warning: return .warningOrange
            case .danger: return .dangerRed
            case .offline: return .textSecondary
            }
        }
        
        var indicator: String {
            switch self {
            case .protected: return "🟢"
            case .warning: return "⚠️"
            case .danger: return "🔴"
            case .offline: return "⚫"
            }
        }
    }
    
    // MARK: - Init
    
    init(
        name: String,
        role: FamilyRole,
        avatar: String,
        status: ProtectionStatus,
        threatsBlocked: Int,
        lastActive: String = "Сейчас",
        action: @escaping () -> Void
    ) {
        self.name = name
        self.role = role
        self.avatar = avatar
        self.status = status
        self.threatsBlocked = threatsBlocked
        self.lastActive = lastActive
        self.action = action
    }
    
    // MARK: - Body
    
    var body: some View {
        Button(action: {
            let generator = UIImpactFeedbackGenerator(style: .medium)
            generator.impactOccurred()
            action()
        }) {
            HStack(spacing: Spacing.m) {
                // Аватар
                Text(avatar)
                    .font(.system(size: Size.iconMedium))
                    .frame(width: Size.avatarSize, height: Size.avatarSize)
                    .background(
                        Circle()
                            .fill(
                                LinearGradient(
                                    colors: [
                                        Color.primaryBlue.opacity(0.3),
                                        Color.primaryBlue.opacity(0.1)
                                    ],
                                    startPoint: .topLeading,
                                    endPoint: .bottomTrailing
                                )
                            )
                    )
                    .overlay(
                        Circle()
                            .stroke(status.color, lineWidth: 3)
                    )
                
                // Информация
                VStack(alignment: .leading, spacing: Spacing.xs) {
                    // Имя и роль
                    HStack(spacing: Spacing.xs) {
                        Text(name)
                            .font(.h3)
                            .foregroundColor(.textPrimary)
                        
                        Text(role.icon)
                            .font(.caption)
                    }
                    
                    // Роль
                    Text(role.label)
                        .font(.caption)
                        .foregroundColor(.textSecondary)
                    
                    // Статистика
                    HStack(spacing: Spacing.m) {
                        // Угрозы заблокированы
                        HStack(spacing: Spacing.xxs) {
                            Text("🛡️")
                                .font(.caption)
                            Text("\(threatsBlocked)")
                                .font(.captionBold)
                                .foregroundColor(.successGreen)
                        }
                        
                        // Последняя активность
                        HStack(spacing: Spacing.xxs) {
                            Text("⏰")
                                .font(.caption)
                            Text(lastActive)
                                .font(.caption)
                                .foregroundColor(.textSecondary)
                        }
                    }
                }
                
                Spacer()
                
                // Статус
                VStack(spacing: Spacing.xs) {
                    Text(status.indicator)
                        .font(.system(size: 24))
                    
                    Text(status.label)
                        .font(.captionSmall)
                        .foregroundColor(status.color)
                }
            }
            .padding(Spacing.cardPadding)
            .background(
                LinearGradient.cardGradient
                    .overlay(Color.white.opacity(0.05))
            )
            .cornerRadius(CornerRadius.large)
            .overlay(
                RoundedRectangle(cornerRadius: CornerRadius.large)
                    .stroke(Color.white.opacity(0.1), lineWidth: 1)
            )
            .cardShadow()
        }
        .buttonStyle(PlainButtonStyle())
    }
}

// MARK: - Preview

#Preview {
    VStack(spacing: Spacing.m) {
        // Родитель - защищён
        FamilyMemberCard(
            name: "Сергей",
            role: .parent,
            avatar: "👨",
            status: .protected,
            threatsBlocked: 47
        ) {
            print("Открыть профиль Сергея")
        }
        
        // Ребёнок - предупреждение
        FamilyMemberCard(
            name: "Маша",
            role: .child,
            avatar: "👧",
            status: .warning,
            threatsBlocked: 23,
            lastActive: "5 мин назад"
        ) {
            print("Открыть профиль Маши")
        }
        
        // Пожилой - оффлайн
        FamilyMemberCard(
            name: "Бабушка",
            role: .elderly,
            avatar: "👵",
            status: .offline,
            threatsBlocked: 12,
            lastActive: "2 часа назад"
        ) {
            print("Открыть профиль Бабушки")
        }
    }
    .padding()
    .background(LinearGradient.backgroundGradient)
}

File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/ui/components/cards/FamilyMemberCard.kt
package family.aladdin.android.ui.components.cards

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import family.aladdin.android.ui.theme.*

/**
 * 👤 Family Member Card
 * Карточка члена семьи
 * Источник: iOS FamilyMemberCard.swift
 */

enum class FamilyRole(val label: String, val icon: String) {
    PARENT("Родитель", "👨‍💼"),
    CHILD("Ребёнок", "👶"),
    TEENAGER("Подросток", "🧒"),
    ELDERLY("Пожилой", "👴")
}

enum class ProtectionStatus(val label: String, val color: Color, val indicator: String) {
    PROTECTED("Защищён", SuccessGreen, "🟢"),
    WARNING("Внимание", WarningOrange, "⚠️"),
    DANGER("Угроза", DangerRed, "🔴"),
    OFFLINE("Оффлайн", TextSecondary, "⚫")
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun FamilyMemberCard(
    name: String,
    role: FamilyRole,
    avatar: String,
    status: ProtectionStatus,
    threatsBlocked: Int,
    lastActive: String = "Сейчас",
    modifier: Modifier = Modifier,
    onClick: () -> Unit = {}
) {
    Card(
        onClick = onClick,
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(CornerRadius.Large),
        colors = CardDefaults.cardColors(
            containerColor = Color.Transparent
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 4.dp
        )
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    brush = Brush.linearGradient(
                        colors = listOf(
                            GradientMiddle.copy(alpha = 0.7f),
                            GradientEnd.copy(alpha = 0.7f)
                        )
                    )
                )
                .padding(Spacing.CardPadding)
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(Spacing.M),
                verticalAlignment = Alignment.CenterVertically
            ) {
                // Аватар с обводкой по статусу
                Box(
                    modifier = Modifier
                        .size(Size.AvatarSize)
                        .background(
                            brush = Brush.linearGradient(
                                colors = listOf(
                                    PrimaryBlue.copy(alpha = 0.3f),
                                    PrimaryBlue.copy(alpha = 0.1f)
                                )
                            ),
                            shape = CircleShape
                        )
                        .border(3.dp, status.color, CircleShape),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = avatar,
                        style = MaterialTheme.typography.displayMedium
                    )
                }
                
                // Информация
                Column(
                    modifier = Modifier.weight(1f),
                    verticalArrangement = Arrangement.spacedBy(Spacing.XS)
                ) {
                    // Имя и роль
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(Spacing.XS)
                    ) {
                        Text(
                            text = name,
                            style = MaterialTheme.typography.displaySmall,
                            color = TextPrimary
                        )
                        Text(
                            text = role.icon,
                            style = MaterialTheme.typography.bodySmall
                        )
                    }
                    
                    // Роль
                    Text(
                        text = role.label,
                        style = MaterialTheme.typography.bodySmall,
                        color = TextSecondary
                    )
                    
                    // Статистика
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(Spacing.M)
                    ) {
                        Row(horizontalArrangement = Arrangement.spacedBy(Spacing.XXS)) {
                            Text(text = "🛡️", style = MaterialTheme.typography.bodySmall)
                            Text(
                                text = "$threatsBlocked",
                                style = MaterialTheme.typography.bodyMedium,
                                color = SuccessGreen
                            )
                        }
                        
                        Row(horizontalArrangement = Arrangement.spacedBy(Spacing.XXS)) {
                            Text(text = "⏰", style = MaterialTheme.typography.bodySmall)
                            Text(
                                text = lastActive,
                                style = MaterialTheme.typography.bodySmall,
                                color = TextSecondary
                            )
                        }
                    }
                }
                
                // Статус
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(Spacing.XS)
                ) {
                    Text(
                        text = status.indicator,
                        style = MaterialTheme.typography.displayMedium
                    )
                    Text(
                        text = status.label,
                        style = MaterialTheme.typography.labelSmall,
                        color = status.color
                    )
                }
            }
        }
    }
}



